hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/ReadableTokenStoreImpl.java

```diff
@@ -87,7 +86,7 @@ private TokenMetadata tokenMetaFrom(final Token token) {
                 token.pauseKeyOrElse(null),
                 token.symbol(),
                 hasRoyaltyWithFallback,
-                token.treasuryAccountNumber(), // remove this and make it a long
+                token.treasuryAccountId().accountNum(), // remove this and make it a long
```

Comment:
Comment seems reverse of intended, perhaps remove it?

Reply:
We should change this tokenMeta to have `treasuryAccountId` instead of long. We are planning to remove `TokenMeta` and use `Token` everywhere

Reply:
Updated, thanks.